File: src/com/laamella/intellij6502/Intellij6502FileTypeFactory.java
package com.laamella.intellij6502;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import com.laamella.intellij6502.asm.AssemblyFileType;
import com.laamella.intellij6502.sprites.SpritesProjectFileType;
import org.jetbrains.annotations.NotNull;

public class Intellij6502FileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
        fileTypeConsumer.consume(AssemblyFileType.INSTANCE);
        fileTypeConsumer.consume(SpritesProjectFileType.INSTANCE);
    }
}

File: src/main/kotlin/com/laamella/intellij6502/Intellij6502FileTypeFactory.kt
package com.laamella.intellij6502

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory
import com.laamella.intellij6502.asm.AssemblyFileType

class Intellij6502FileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        fileTypeConsumer.consume(AssemblyFileType.INSTANCE)
    }
}
